### DISCUSSION TITLE
How to export to IndexedDB?

### DISCUSSION THREAD

**Discussion by maxzhuo** [duckdb/duckdb-wasm#1689] [Q&A]
How does DuckDB-wasm save a database to IndexedDB and reload the database in IndexedDB after refreshing the page? Please write a code example, thanks.

**Comment by ravwojdyla** [comment]
@maxzhuo see https://github.com/duckdb/duckdb-wasm/pull/1490#issuecomment-2036915766 for some summary of relevant past discussions and the PR itself is directly relevant.

**Reply by maxzhuo** [reply]
I tried running it on the console：

```js
await db.open({
    path: 'opfs://test.db',
    accessMode: duckdb.DuckDBAccessMode.READ_WRITE,
});
const conn = await db.connect();
```

After running, two warnings were displayed：

⚠️ ▷ Buffering missing file: opfs://test.db
⚠️ ▷ Buffering missing file: opfs://test.db.wal

**Reply by maxzhuo** [reply]
Is test.db automatically created?
Do I need an additional JavaScript code to create test.db in advance?
